File: memorizer_android/app/src/main/java/cz/dodo/memorizer/entities/SpeciesItem.kt
package cz.dodo.memorizer.entities

import paperparcel.PaperParcel
import paperparcel.PaperParcelable
import java.io.Serializable

@PaperParcel
data class SpeciesItem(val name: LocString, val images: List<String>)  : PaperParcelable, Serializable {
    companion object {
        @JvmField
        val CREATOR = PaperParcelSpeciesItem.CREATOR
    }
}

File: memorizer_flutter/memorizer/lib/entities/species_item.dart

import 'package:memorizer/entities/loc_string.dart';

class SpeciesItem extends Object {
  final LocString name;
  final List<String> imageUrls;

  String get imageUrl {
    return imageUrls.first;
  }

  SpeciesItem.fromJSON(Map<String, dynamic> json)
      : name = LocString.fromJSON(json['name']),
        imageUrls= (json['images'] as List).map((dyn) => dyn as String).toList();

}